build/deploy

```diff
@@ -3,6 +3,7 @@
 git checkout master && \
   git pull origin master && \
   git checkout gh-pages && \
+  git pull origin gh-pages && \
```

Comment:
Are we sure we want to potentially create merges with these two `pull` commands? How about `pull --ff-only` for both of these?

Reply:
For `gh-pages` maybe this should be `git reset --hard origin/gh-pages`, since nobody should ever be committing to `gh-pages` except to deploy, so nobody should care about local changes?

Regarding master, depends who's going to be deploying I guess? Since the fix would be to rebase master. Presumably the only way you could fail with `pull --ff-only` is if you'd committed locally and not pushed those changes?